### PR TITLE
feat: calculate percentiles using different library (montanaflynn/stats)

### PR DESCRIPTION
**Pull request by scott-david-walker**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [x] added new dependencies
- [ ] updated the docs
- [x] added a test

## Breaking changes

-

## Changes

- Use new library to calculate percentiles (montanaflynn/stats)
- Added unit tests

## Fixes

- #2994

### CODE REVIEW

File: internal/pkg/api/repository/common/metrics.go [hunk 1]
```diff
@@ -52,5 +48,12 @@ func CalculateMetrics(executionsMetrics []testkube.ExecutionsMetricsExecutions)
 	metrics.ExecutionDurationP99ms = int32(durationP99 / time.Millisecond)
 
 	return
+}
 
+func calculate(durations []float64, quantile float64) float64 {
+	percentile, err := stats.PercentileNearestRank(durations, quantile)
+	if err != nil {
+		return 0
```
Comment (vsukhin) [inline internal/pkg/api/repository/common/metrics.go:56]: log error just in case something unexpected happen. good job!

**Review by vsukhin — APPROVED** [review]

**Review by vLia — APPROVED** [review]
Looks good, always excited to see new tests added! 👍

**Comment by vsukhin** [comment]
@scott-david-walker check unit tests
    --- FAIL: TestTestExecutionsMetrics/getting_percentiles_of_execution_duration (0.00s)
FAIL
FAIL

and run go fmt for your changes

File: internal/pkg/api/repository/common/metrics.go (hunk 1 above)
Comment (scott-david-walker) [reply internal/pkg/api/repository/common/metrics.go:58]: fixed

**Comment by scott-david-walker** [comment]
Could someone give me advice on what's going on with the linting error? 

`Cannot open: File exists` in [this action](https://github.com/kubeshop/testkube/actions/runs/3988917806/jobs/6841764847)
I've ran the gofmt commands so I'm at a bit of a loss

Apologies if this is a stupid question, bit of a noob at GoLang :-)

**Comment by vsukhin** [comment]
@scott-david-walker make sure your code is go formatted. in other case, it could be github caching issue

**Comment by vsukhin** [comment]
@scott-david-walker ignore linting check,  just solve merging issue. it should be enough for merging

**Pull request merged**